Shopify/cli
# [Themes] Remove cli-ruby from cli-kit

### WHY are these changes introduced?

Fixes part of https://github.com/Shopify/develop-advanced-edits/issues/364

Follow-up is tracked in the same issue (docs, analytics, etc)

### WHAT is this pull request doing?
- Removes `cli-ruby` from the `cli-kit`
- This should also cause autogenerated documentation in `docs/api/cli-kit` for `node-ruby` to be removed
<img width="303" alt="image" src="https://github.com/user-attachments/assets/813a9c55-6a38-4045-bbbf-e203e69d6563">


### How to test your changes?
- CI should cover these changes
- Run an `app` command
- Run a `theme` command

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
